gridcapa-cse-valid-app/src/main/java/com/farao_community/farao/cse_valid/app/dichotomy/DichotomyRunner.java

```diff
@@ -52,33 +55,47 @@ public DichotomyRunner(FileImporter fileImporter,
         this.cseValidNetworkShifter = cseValidNetworkShifter;
     }
 
-    public DichotomyResult<RaoResponse> runImportCornerDichotomy(TTimestampWrapper timestampWrapper, CseValidRequest cseValidRequest, String jsonCracUrl, String raoParametersURL, Network network) {
-        int npAugmented = timestampWrapper.getMniiIntValue();
-        int np = timestampWrapper.getMibniiIntValue() - timestampWrapper.getAntcfinalIntValue();
-        double maxValue = (double) npAugmented - np;
-        businessLogger.info(DICHOTOMY_PARAMETERS_MSG, DEFAULT_MIN_INDEX, (int) maxValue, (int) DEFAULT_DICHOTOMY_PRECISION);
-        DichotomyEngine<RaoResponse> engine = new DichotomyEngine<>(
-                new Index<>(DEFAULT_MIN_INDEX, maxValue, DEFAULT_DICHOTOMY_PRECISION),
-                INDEX_STRATEGY_CONFIGURATION,
-                cseValidNetworkShifter.getNetworkShifterForFullImport(timestampWrapper, network, cseValidRequest.getGlsk().getUrl(), cseValidRequest.getProcessType()),
-                getNetworkValidator(cseValidRequest, jsonCracUrl, raoParametersURL));
+    public DichotomyResult<RaoResponse> runDichotomy(TTimestampWrapper timestampWrapper,
+                                                     CseValidRequest cseValidRequest,
+                                                     String jsonCracUrl,
+                                                     String raoParametersURL,
+                                                     Network network,
+                                                     boolean isForExportCorner) {
+        final double minValue;
+        final double maxValue;
+        final NetworkShifter networkShifter;
+        if (isForExportCorner) {
+            Network initialNetwork = fileImporter.importNetwork(cseValidRequest.getCgm().getUrl());
+            double franceImportBeforeShifting = NetPositionHelper.computeFranceImportFromItaly(initialNetwork);
+            double franceImportAfterShifting = NetPositionHelper.computeFranceImportFromItaly(network);
+            if (timestampWrapper.isFranceImportingFromItaly()) {
+                minValue = franceImportBeforeShifting - franceImportAfterShifting;
+                maxValue = DEFAULT_MAX_INDEX;
```

Comment:
The `maxValue` is the same in the `if` and `else` branches. You should move this assignment outside the conditional statement.

Reply:
And maybe it will be possible to simplify the if-else after that.